src/SigninResponse.ts

```diff
@@ -29,31 +29,29 @@ export class SigninResponse {
 
     // set by ResponseValidator
     /** custom "state", which can be used by a caller to have "data" round tripped */
-    public state: unknown | undefined;
+    public state: unknown;
 
     // set by ResponseValidator
-    public profile: UserProfile;
-
-    public constructor(url?: string, delimiter = "#") {
-        const values = UrlUtils.parseUrlFragment(url, delimiter);
-
-        this.error = values.error;
-        this.error_description = values.error_description;
-        this.error_uri = values.error_uri;
-
-        this.code = values.code;
-        this.state_id = values.state;
-
-        this.id_token = values.id_token;
-        this.session_state = values.session_state;
-        this.access_token = values.access_token;
-        this.refresh_token = values.refresh_token;
-        this.token_type = values.token_type;
-        this.scope = values.scope;
-        this.expires_in = parseInt(values.expires_in);
-
-        this.state = undefined;
-        this.profile = {};
+    public profile: UserProfile = {};
+
+    public constructor(params: URLSearchParams) {
+        // URLSearchParams returns `null` for missing values, reconstruct it as a map
+        const values = new Map(params);
+        this.error = values.get("error");
+        this.error_description = values.get("error_description");
+        this.error_uri = values.get("error_uri");
+
+        this.code = values.get("code")!;
```

Comment:
Should I refactor this to throw an error if these required params aren't found? Or should I add exceptions for the lint warning?

Reply:
Technically it would be possible, practically it wont happen.

Analog to other code parts i would throw an error, its better than suppress lint warnings

Reply:
i saw you added eslint suppress comments. Options i see:
- eslint suppress
- values.get("code") || "" (probably the best)
- throw error (very explicit, but hard)

Reply:
I looked into this a bit further, it looks like we depend on ResponseValidator to clean up the bad responses. I'll default them to empty strings since I also agree that it's the least hacky.